Lib/test/test_shutil.py

```diff
@@ -1665,6 +1665,49 @@ def test_register_archive_format(self):
         formats = [name for name, params in get_archive_formats()]
         self.assertNotIn('xxx', formats)
 
+    def _unlink_existing_file(self, path):
+        try:
+            os.unlink(path)
+        except FileNotFoundError:
+            pass
+
+    def test_make_tarfile_rootdir_nodir(self):
+        # GH-99203
+        self.addCleanup(self._unlink_existing_file, f'{TESTFN}.tar')
+        for dry_run in (0, True):
```

Comment:
Why not False?

Reply:
`make_archive`'s default value for `dry_run` is `0` while according to the [docs](https://docs.python.org/3/library/shutil.html#shutil.make_archive) dry run mode is enabled with `True`. The implementation checks truthy/falsiness though. I've changed the tests to use `False`/`True`.